dart/biomechanics/OpenSimParser.cpp

```diff
@@ -3064,67 +3064,52 @@ std::vector<ForcePlate> OpenSimParser::loadGRF(
   assert(timestamps.size() == copRows.size());
   assert(timestamps.size() == wrenchRows.size());
 
-  int downsampleByFactor = 1;
-  if (timestamps.size() > 1)
+  // Check that the values in timestamps cover the range of values in
+  // targetTimestamps
+  if (!targetTimestamps.empty()) {
+    assert(timestamps.size() > 0);
```

Comment:
These asserts will all be stripped out in release builds, and we'll just get hard-to-debug segfaults if we pass bad data. Perhaps if each of these checks fail, we print to std::out and then throw an exception or return an exit code?